src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -106,18 +106,19 @@ internal DeconstructionUncommonData(DeconstructMethodInfo deconstructMethodInfoO
 
         private sealed class CollectionExpressionUncommonData : NestedUncommonData
         {
-            internal CollectionExpressionUncommonData(CollectionExpressionTypeKind collectionExpressionTypeKind, TypeSymbol? elementType, ImmutableArray<Conversion> elementConversions) :
+            internal CollectionExpressionUncommonData(CollectionExpressionTypeKind collectionExpressionTypeKind, TypeSymbol elementType, ImmutableArray<Conversion> elementConversions) :
                 base(elementConversions)
             {
+                Debug.Assert(collectionExpressionTypeKind != CollectionExpressionTypeKind.None);
```

Comment:
consider adding assertion that `elementType` is not null (we already have such assertion in caller, but I think it would make sense here)